ipatests/test_integration/test_bugzilla.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) 2015  FreeIPA Contributors see COPYING for license
+#
+
+import time
+from tempfile import NamedTemporaryFile
+
+from ipatests.pytest_plugins.integration import tasks
+from ipatests.pytest_plugins.integration.env_config import get_global_config
+from ipatests.test_integration.base import IntegrationTest
+
+config = get_global_config()
+
+
+class LayoutsBaseTest(IntegrationTest):
+    @classmethod
+    def install(cls, mh):
+        # tests use custom installation
+        pass
+
+    def replication_is_working(self):
+        test_user = 'replication-testuser'
+        self.master.run_command(
+            ['ipa', 'user-add', test_user, '--first', 'test', '--last', 'user']
+        )
+
+        time.sleep(60)  # make sure the replication of user is done
+
+        for r in self.replicas:
+            r.run_command(['ipa', 'user-show', test_user])
+
+
+class TestReplicaInstallWithExistingEntry(LayoutsBaseTest):
+    """Automation for:
+        https://bugzilla.redhat.com/show_bug.cgi?id=1493145
+        https://bugzilla.redhat.com/show_bug.cgi?id=1506913
+        ipa-replica-install might fail because of an already existing
+        entry cn=ipa-http-delegation,cn=s4u2proxy,cn=etc,$SUFFIX"""
+    num_replicas = 1
+
+    def test_replica_install_with_existing_entry(self):
+        result = tasks.install_master(self.master)
+        assert result.returncode == 0
+        master = self.master
+        replica = self.replicas[0]
+        tf = NamedTemporaryFile()
+        ldif_file = tf.name
+        base_dn = "dc=%s" % (",dc=".join(replica.domain.name.split(".")))
+        entry_ldif = """
+        dn: cn=ipa-http-delegation,cn=s4u2proxy,cn=etc,%s\n
+        changetype: modify\n
+        add: memberPrincipal\n
+        memberPrincipal: HTTP/%s@%s\n
+
+        dn: cn=ipa-ldap-delegation-targets,cn=s4u2proxy,cn=etc,%s\n
+        changetype: modify\n
+        add: memberPrincipal\n
+        memberPrincipal: ldap/%s@%s""" % (
+        base_dn, replica.hostname, replica.domain.name.upper(),
+        base_dn, replica.hostname, replica.domain.name.upper())
+        master.put_file_contents(ldif_file, entry_ldif)
+        arg = ['ldapmodify', '-h', master.hostname, '-p', '389',
```

Comment:
It also a deserves a comment what is the intention as the next reader will most likely not know. Somethink like:

This simulates a situation where `ipa server-del` is not called on master but only as part of replica uninstallation and where there deletions of `ipa-http-delegation,cn=s4u2proxy` entries are not propagated to master.

Reply:
@stlaz  or other, do you know if we already have some tasks which would do LDAP modify in tests? Seems like a common task to me.